src/cli/helpers.ts

```diff
@@ -85,6 +100,36 @@ export function orderPickles<T = string>(
         shuffle(pickleIds, seed)
       }
       break
+    case 'rerun':
+      {
+        if (unexpandedFeaturePaths === undefined) {
+          throw new Error(
+            'Cannot order by rerun because no unexpandedFeaturePaths were provided'
+          )
+        }
+
+        const picklesWithDocument =
+          pickleIds as unknown[] as PickleWithDocument[]
+
+        picklesWithDocument.sort((a, b) => {
+          const pathA = `${a.pickle.uri}:${a.location.line}`
+          const pathB = `${b.pickle.uri}:${b.location.line}`
+          const indexA = unexpandedFeaturePaths.indexOf(pathA)
+          const indexB = unexpandedFeaturePaths.indexOf(pathB)
+          if (indexA === -1) {
+            throw new Error(
+              `Cannot use rerun order because ${pathA} was not in the rerun order. Did you forget to specify @rerun.txt?`
+            )
+          }
+          if (indexB === -1) {
+            throw new Error(
+              `Cannot use rerun order because ${pathB} was not in the rerun order. Did you forget to specify @rerun.txt?`
+            )
+          }
+          return indexA - indexB
```

Comment:
Note I think this could be something like

```js
// validate unexpandedFeaturePaths has no duplicates

const ordered = new Array(unexpandedFeaturePaths.length);
picklesWithDocument.forEach(item => {
  // get index in unexpandedFeaturePaths, throw if not found
  ordered[index] = item
})

// validate ordered has not undefined elements (not sure if needed)
```

That should be O(n) instead O(nlogn)